backends/p4tools/p4check/commands.cpp

```diff
@@ -0,0 +1,177 @@
+#include "backends/p4tools/p4check/commands.h"
+
+#include <string.h>
+
+#include <algorithm>
+#include <climits>
+#include <map>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include <boost/none.hpp>
+
+#include "lib/exceptions.h"
+#include "lib/map.h"
+
+namespace P4Tools {
+
+P4CheckCommands::P4CheckCommands() {
+    registerCommand(HELP, {}, {"help", "-h", "--help"}, "Shows this help message and exits");
+    registerCommand(VERSION, {}, {"version", "--version"}, "Prints version information and exits");
+    registerCommand(TESTGEN, {"p4testgen"}, {"testgen"}, "Generates packet tests for P4 programs");
+    registerCommand(SMITH, {"smith"}, {"smith"}, "Generates random p4 programs");
```

Comment:
are these command implemented for any backend?

Reply:
They will be added later as part of new pull requests. Currently these should throw an error, if called.